src/Handler/Table/Import/ImportTableFromTableHandler.php

```diff
@@ -219,32 +220,52 @@ private function import(
             return [null, $importState->getResult()];
         }
 
-        $stagingTable = $this->createStageTable($destinationDefinition, $sourceMapping, $bqClient);
+        /** @var TableImportFromTableCommand\SourceTableMapping\ColumnMapping[] $mappings */
+        $mappings = iterator_to_array($sourceMapping->getColumnMappings()->getIterator());
+        $stagingTable = StageTableDefinitionFactory::createStagingTableDefinitionWithMapping(
+            $destinationDefinition,
+            $mappings,
+        );
+        // if column names are identical, we can use CREATE TABLE COPY (CopyImportFromTableToTable)
+        // if column names are not identical, we have to use INSERT INTO (ToStageImporter)
+        $isColumnNameIdentical = true;
+        foreach ($mappings as $columnMapping) {
+            if ($columnMapping->getSourceColumnName() === $columnMapping->getDestinationColumnName()) {
+                continue;
+            }
+            $isColumnNameIdentical = false;
+        }
```

Comment:
prakticky pokud ty sloupce nepotřebujů rename tak to ide přes copy. Přemýšlal sem o tom, že bych bych ty sloupce aj přejmenovával po copy, ale to mě přišlo jako overkill, on stejně IM pro BQ není takže ten rename sloupců tam byt teď nemože.
Stejně tak aj pokud by sa měnil typ u netypových tabulek tak by to nefakčilo, ale to taky teď nejde.